pkg/kn/commands/types.go

```diff
@@ -27,10 +31,28 @@ var CfgFile string
 // KubeCfgFile is the path for the Kubernetes config
 var KubeCfgFile string
 
+// Extend v1alpha1 serving client by a method for updating GVK
+type ServingV1alpha1InterfaceWithGvkUpdater interface {
```

Comment:
if we are wrapping ServingV1alpha1Interface why not set GVK transparently in its methods?

Reply:
Yes, we could do that, too (IMO depends on how many methods to wrap, though, whether that is suitable. Haven't checked yet). 

For the alternative is to have a more custom client library like approach suited for kn as provided in https://github.com/rhuss/client/blob/pr/get/pkg/serving/client.go (especially for cross-cutting methods like `RevisionsForService` as in https://github.com/rhuss/client/blob/744872f9350f29f1cdf09ba21a7b732a53cece53/pkg/serving/client.go#L56-L69)
which I'm going to propose towards the end of the week.

Reply:
(sorry that this comment was written under another account as I was logged in accidentally with my bots GitHub account at that time)